pkg/apis/aks.cattle.io/v1/types.go

```diff
@@ -0,0 +1,87 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AKSClusterConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AKSClusterConfigSpec   `json:"spec"`
+	Status AKSClusterConfigStatus `json:"status"`
+}
+
+// AKSClusterConfigSpec is the spec for a AKSClusterConfig resource
+type AKSClusterConfigSpec struct {
+	DisplayName                 string            `json:"displayName" norman:"noupdate"`
+	Imported                    bool              `json:"imported" norman:"noupdate"`
+	ResourceLocation            string            `json:"resourceLocation" norman:"noupdate"`
+	ResourceGroup               string            `json:"resourceGroup" norman:"noupdate"`
+	ClusterName                 string            `json:"clusterName" norman:"noupdate"`
+	BaseURL                     string            `json:"baseUrl"`
+	AuthBaseURL                 string            `json:"authBaseUrl"`
+	SubscriptionID              string            `json:"subscriptionId"`
+	TenantID                    string            `json:"tenantId"`
+	AzureCredentialSecret       string            `json:"azureCredentialSecret"`
+	NetworkPlugin               string            `json:"networkPlugin" norman:"noupdate"`
+	VirtualNetworkResourceGroup string            `json:"virtualNetworkResourceGroup" norman:"noupdate"`
+	VirtualNetwork              string            `json:"virtualNetwork" norman:"noupdate"`
+	Subnet                      string            `json:"subnet" norman:"noupdate"`
+	NetworkDNSServiceIP         string            `json:"dnsServiceIp" norman:"noupdate"`
+	NetworkServiceCIDR          string            `json:"serviceCidr" norman:"noupdate"`
+	NetworkDockerBridgeCIDR     string            `json:"dockerBridgeCidr" norman:"noupdate"`
+	NetworkPodCIDR              string            `json:"podCidr" norman:"noupdate"`
+	LoadBalancerSKU             string            `json:"loadBalancerSku" norman:"noupdate"`
+	NetworkPolicy               string            `json:"networkPolicy" norman:"noupdate"`
+	LinuxAdminUsername          string            `json:"adminUsername,omitempty" norman:"noupdate"`
+	LinuxSSHPublicKeyContents   string            `json:"sshPublicKeyContents,omitempty" norman:"noupdate"`
+	DNSPrefix                   *string           `json:"dnsPrefix,omitempty" norman:"noupdate"`
+	KubernetesVersion           *string           `json:"kubernetesVersion"`
+	Tags                        map[string]string `json:"tags"`
+	NodePools                   []AKSNodePool     `json:"nodePools"`
+	PrivateCluster              *bool             `json:"privateCluster" norman:"noupdate"`
+	AuthorizedIPRanges          *[]string         `json:"authorizedIpRanges"`
+}
```

Comment:
It looks like a lot of feedback has been addressed I just wanted to document and make it clear that all top level fields here that map to a cluster field should be pointers. Then during update these should be skipped if they are not nil. In validation non-imported clusters should not have any fields as nil. If you have any questions regarding this still ask me, it is also explained in the video during the syncing part, or in the EKS docs.

Reply:
I have changed cluster options to pointers, as I understand pointers should be only options which can be updated for cluster configuration. With AKS it looks that only few options can changed once cluster was provisioned